fineract-provider/src/main/java/org/apache/fineract/cob/loan/SetLoanDelinquencyTagsBusinessStep.java

```diff
@@ -55,7 +67,17 @@ public Loan execute(Loan loan) {
                 // the
                 // current date and not the previous (COB) date.
                 ThreadLocalContextUtil.setActionContext(ActionContext.DEFAULT);
-                loanAccountDomainService.setLoanDelinquencyTag(loan, DateUtils.getBusinessLocalDate());
+
+                final List<LoanDelinquencyAction> savedDelinquencyList = delinquencyReadPlatformService
+                        .retrieveLoanDelinquencyActions(loan.getId());
+                List<LoanDelinquencyActionData> effectiveDelinquencyList = delinquencyEffectivePauseHelper
+                        .calculateEffectiveDelinquencyList(savedDelinquencyList);
+
+                if (!isDelinquencyOnPause(loan, effectiveDelinquencyList)) {
+                    loanAccountDomainService.setLoanDelinquencyTag(loan, DateUtils.getBusinessLocalDate(), effectiveDelinquencyList);
+                } else {
+                    log.info("Delinquency is on pause for loan with ID [{}]", loan.getId());
```

Comment:
We shall not log this on info imo. We can have hundreds of loans like this and that would generate noise.